evm/src/main/java/org/hyperledger/besu/evm/processor/ModifiablePrecompilesMessageCallProcessor.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright contributors to Besu.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.evm.processor;
+
+import org.hyperledger.besu.datatypes.Address;
+import org.hyperledger.besu.evm.precompile.PrecompileContractRegistry;
+
+import java.util.Map;
+
+/**
+ * A message call processor designed specifically for simulation purposes that allows for overriding
+ * precompile addresses.
+ */
+public class ModifiablePrecompilesMessageCallProcessor extends MessageCallProcessor {
```

Comment:
Name seems a bit odd to me, maybe rename to `OverriddenPrecompilesMessageCallProcessor`